test/core/formattersTests.ts

```diff
@@ -1,193 +1,280 @@
-    });
+  describe("currency()", () => {
+    it("formats values based on precision, symobl, and prefix correctly", () => {
+      let defaultFormatter = Plottable.Formatters.currency();
+      assert.strictEqual(defaultFormatter(1), "$1.00", "formatted correctly with default \"$\" prefix");
+      assert.strictEqual(defaultFormatter(-1), "-$1.00", "formatted negative value correctly with default \"$\" prefix");
+      assert.strictEqual(defaultFormatter(1.999), "$2.00", "formatted correctly with default \"$\" prefix");
+      assert.strictEqual(defaultFormatter(-1.234), "-$1.23", "formatted negative value correctly with default \"$\" prefix");
+
+      let currencyFormatter0 = Plottable.Formatters.currency(0);
+      assert.strictEqual(currencyFormatter0(1.1234), "$1", "formatted with correct precision");
+
+      let poundFormatter = Plottable.Formatters.currency(2, "£");
+      assert.strictEqual(poundFormatter(1.1234), "£1.12", "formatted with correct precision and \"£\" as prefix");
+      assert.strictEqual(poundFormatter(-100.987), "-£100.99", "formatted with correct precision and \"£\" as prefix");
 
-    it("can change the type and position of the currency symbol", () => {
       let centsFormatter = Plottable.Formatters.currency(0, "c", false);
-      let result = centsFormatter(1);
-      assert.strictEqual(result.charAt(result.length - 1), "c", "The specified currency symbol was appended");
+      assert.strictEqual(centsFormatter(1), "1c", "formatted correctly with \"c\" as suffix");
+      assert.strictEqual(centsFormatter(-1), "-1c", "formatted correctly with \"c\" as suffix");
     });
 
-    it("non-integer precision throws exception", () => {
-      assert.throws(() => Plottable.Formatters.currency(2.1)(123.313), "Formatter precision must be an integer");
+    it("throws an error on invalid precision", () => {
+      let nonIntegerValues = [null, 2.1, NaN, "5", "abc"];
+      nonIntegerValues.forEach((value) =>
+        (<any> assert).throws(() => Plottable.Formatters.currency(<any> value), Error,
+          "Formatter precision must be an integer", `${value} is not a valid precision value`)
+      );
+
+      let outOfBoundValues = [-1, 21, -Infinity, Infinity];
+      outOfBoundValues.forEach((value) =>
+        (<any> assert).throws(() => Plottable.Formatters.currency(<any> value), Error,
+          "Formatter precision must be between 0 and 20", `${value} is not a valid precision value`)
+      );
+    });
+
+    it.skip("throws exception for non-numeric values", () => {
+      let nonNumbricValues: any[] = [null, undefined, "123", "abc", ""];
+      // "$0.00", "$NaN", "$123.00", "$NaN", "$0.00"
+      let defaultFormatter = Plottable.Formatters.currency();
+      nonNumbricValues.forEach((value) =>
+        (<any> assert).throws(() => defaultFormatter(<any> value), Error,
+          "", `${value} is not a valid value for Formatter.currency`)
```

Comment:
it's not set because the formatter doesn't handle these special cases.
I feel like determining how error handling should work is outside the scope of this CR
same for below